GameServerLib/GameObjects/Missiles/Projectile.cs

```diff
@@ -64,39 +114,113 @@ public override void Update(float diff)
             base.Update(diff);
         }
 
-        public override void OnCollision(IGameObject collider)
+        public override void OnCollision(IGameObject collider, bool isTerrain = false)
         {
-            base.OnCollision(collider);
-            if (Target != null && Target.IsSimpleTarget && !IsToRemove())
+            if (Destination != Vector2.Zero && !IsToRemove())
             {
                 CheckFlagsForUnit(collider as IAttackableUnit);
             }
             else
             {
-                if (Target == collider)
+                if (TargetUnit == collider)
                 {
                     CheckFlagsForUnit(collider as IAttackableUnit);
                 }
             }
+
+            // TODO: Implement methods for isTerrain for projectiles such as Nautilus Q, ShyvanaDragon Q, or Ziggs Q.
         }
 
         public override float GetMoveSpeed()
         {
             return _moveSpeed;
         }
 
-        // todo refactor this
+        /// <summary>
+        /// Moves this projectile to either its target unit, or its destination, and updates its coordinates along the way.
+        /// </summary>
+        /// <param name="diff">The amount of milliseconds the AI is supposed to move</param>
+        public override void Move(float diff)
+        {
+            if (!HasTarget())
```

Comment:
No, because projectiles don't have waypoints, they only have a target or a destination. This new Move function removes the unneeded complexity of waypoints.